src/Illuminate/Routing/RouteDependencyResolverTrait.php

```diff
@@ -77,7 +78,11 @@ protected function transformDependency(ReflectionParameter $parameter, $paramete
         // the list of parameters. If it is we will just skip it as it is probably a model
         // binding and we do not want to mess with those; otherwise, we resolve it here.
         if ($className && ! $this->alreadyInParameters($className, $parameters)) {
-            return $parameter->isDefaultValueAvailable() ? null : $this->container->make($className);
+            $isEnum = method_exists(ReflectionClass::class, 'isEnum') ? (new ReflectionClass($className))->isEnum() : false;
+
```

Comment:
You could tighten it up with:

```php
$isEnum = method_exists(ReflectionClass::class, 'isEnum') && (new ReflectionClass($className))->isEnum();
```

I think your nested ternary is understandable as-is.

Reply:
@BrandonSurowiec Thanks for the idea, just changed it!